From 66e60c288dacad21797175dc21763ece765338f4 Mon Sep 17 00:00:00 2001
From: Mika Westerberg <mika.westerberg@linux.intel.com>
Date: Fri, 2 May 2014 12:12:45 +0300
Subject: [PATCH 081/152] i2c: designware: Implement runtime PM hooks

We can follow the PCI driver here and re-use system sleep hooks with the
addition that we must disable the HW in case of system sleep.

Signed-off-by: Mika Westerberg <mika.westerberg@linux.intel.com>
---
 drivers/i2c/busses/i2c-designware-platdrv.c |   13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

Index: pc-linux/drivers/i2c/busses/i2c-designware-platdrv.c
===================================================================
--- pc-linux.orig/drivers/i2c/busses/i2c-designware-platdrv.c	2014-07-08 00:25:13.660708323 -0500
+++ pc-linux/drivers/i2c/busses/i2c-designware-platdrv.c	2014-07-08 00:25:36.424707959 -0500
@@ -395,12 +395,13 @@ static const struct of_device_id dw_i2c_
 MODULE_DEVICE_TABLE(of, dw_i2c_of_match);
 #endif
 
-#ifdef CONFIG_PM_SLEEP
+#ifdef CONFIG_PM
 static int dw_i2c_suspend(struct device *dev)
 {
 	struct platform_device *pdev = to_platform_device(dev);
 	struct dw_i2c_dev *i_dev = platform_get_drvdata(pdev);
 
+	i2c_dw_disable(i_dev);
 	clk_disable_unprepare(i_dev->clk);
 
 	return 0;
@@ -418,13 +419,11 @@ static int dw_i2c_resume(struct device *
 
 	return 0;
 }
-
-static SIMPLE_DEV_PM_OPS(dw_i2c_dev_pm_ops, dw_i2c_suspend, dw_i2c_resume);
-#define DW_I2C_DEV_PM_OPS	(&dw_i2c_dev_pm_ops)
-#else
-#define DW_I2C_DEV_PM_OPS	NULL
 #endif
 
+static UNIVERSAL_DEV_PM_OPS(dw_i2c_dev_pm_ops, dw_i2c_suspend,
+			    dw_i2c_resume, NULL);
+
 /* work with hotplug and coldplug */
 MODULE_ALIAS("platform:i2c_designware");
 
@@ -436,7 +435,7 @@ static struct platform_driver dw_i2c_dri
 		.owner	= THIS_MODULE,
 		.of_match_table = of_match_ptr(dw_i2c_of_match),
 		.acpi_match_table = ACPI_PTR(dw_i2c_acpi_match),
-		.pm	= DW_I2C_DEV_PM_OPS,
+		.pm	= &dw_i2c_dev_pm_ops,
 	},
 };
 
